File: Euler031.java

public class p31 {
	public static void main(String[] args){
		int goal = 200;
		int ways = 0;
		for (int a = 0; a<=goal; a+= 200){
			for (int b=a; b<=goal; b+=100){
				for (int c=b;c<=goal;c+=50){
					for (int d=c; d<=goal;d+=20){
						for (int e=d; e<=goal; e+=10){
							for (int f=e; f<=goal; f+=5){
								for (int g=f; g<=goal; g+=2){
											ways++;
								}
							}
						}
					}
				}
			}
		}
		System.out.println(ways);
	}
}

File: Euler031.rb
ways=0
(0..200).step(200) do |a|
	(a..200).step(100) do |b|
		(b..200).step(50) do |c|
			(c..200).step(20) do |d|
				(d..200).step(10) do |e|
					(e..200).step(5) do |f|
						(f..200).step(2) do |g|
							ways+=1
						end
					end
				end
			end
		end
	end
end
puts ways

	